### PR TITLE
Fix/Disable file uploads larger than 20 MB

### LINKED ISSUE #264: >20mb files can't be uploaded
Currently SDK allows to upload any file, but uploading stop if file is bigger than 20mb. And error message doesn't say anything about it. So it might be annoying especially with slow network connection.

Possible solution: when user tries to upload file we check file size and tell user that currently it's not possible to upload the file, because file limit is 20mb.

### PR DESCRIPTION
**Pull request by adrian09h**
# Submit a pull request
close #264 

image

image

### CODE REVIEW

File: library/src/main/java/com/getstream/sdk/chat/model/Attachment.java [hunk 1]
```diff
@@ -294,6 +296,14 @@ public int getProgress() {
         public void setProgress(int progress) {
             this.progress = progress;
         }
+
+        public boolean isFileTooLarge() {
+            return fileTooLarge;
+        }
+
+        public void setFileTooLarge(boolean fileTooLarge) {
```
Comment (tbarbugli) [inline library/src/main/java/com/getstream/sdk/chat/model/Attachment.java]: this field is useless

File: library/src/main/java/com/getstream/sdk/chat/model/Attachment.java [hunk 2]
```diff
@@ -294,6 +296,14 @@ public int getProgress() {
         public void setProgress(int progress) {
             this.progress = progress;
         }
+
+        public boolean isFileTooLarge() {
+            return fileTooLarge;
```
Comment (tbarbugli) [inline library/src/main/java/com/getstream/sdk/chat/model/Attachment.java]: a file is too large when its size is > than max size, no need to rely on a private boolean for that, just change this into the expression

Comment (adrian09h) [reply library/src/main/java/com/getstream/sdk/chat/model/Attachment.java]: Yeah, exactly, I know it either.
However if we don't use `setter` and `getter` functions for `fileTooLarge` in Config of `Attachment`  we need some messy codes to check file size exceeds the maxim size.

Each time first we have to get file from file path and then have to compare with limit max upload size.

If the part of checking file size is one or two, it wouldn't problem. unfortunately we have to check this in here and there at least 4 times more.

Comment (tbarbugli) [reply library/src/main/java/com/getstream/sdk/chat/model/Attachment.java]: cache the result internally then

Comment (adrian09h) [reply library/src/main/java/com/getstream/sdk/chat/model/Attachment.java]: removed.

File: library/src/main/java/com/getstream/sdk/chat/utils/StringUtility.java [hunk 3]
```diff
@@ -66,4 +68,14 @@ public static String convertVideoLength(long videoLength) {
         long seconds = videoLength % 60;
         return String.format("%02d:%02d:%02d", hours, minutes, seconds);
     }
+
+    @SuppressLint("DefaultLocale")
+    public static String convertFileSizeByteCount(long bytes) {
+        int unit = 1000;
+        if (bytes < unit) return bytes + " B";
```
Comment (elevenetc) [inline library/src/main/java/com/getstream/sdk/chat/utils/StringUtility.java]: I would add also zero and negative cases. Never trust input parameters.
That's how negative unread counters appear in production.

File: library/src/main/java/com/getstream/sdk/chat/utils/MessageInputController.java [hunk 4]
```diff
@@ -187,7 +192,11 @@ private void configSelectAttachView(boolean isMedia, List<Attachment> editAttach
                     binding.lvFile.setOnItemClickListener((AdapterView<?> parent, View view,
                                                            int position, long id) -> {
                         Attachment attachment = attachments.get(position);
```
Comment (elevenetc) [inline library/src/main/java/com/getstream/sdk/chat/utils/MessageInputController.java]: Duplicated attachment code, could be exctracted into:
```java
@Nullable
private Attachment getAttachment(List<Attachment> attachments, int position) {
    Attachment attachment = attachments.get(position);
    File file = new File(attachment.config.getFilePath());
    if (file.length() > Constant.MAX_UPLOAD_FILE_SIZE) {
        Utils.showMessage(context, context.getResources().getString(R.string.stream_large_size_file_error));
        return null;
    }
    attachment.config.setSelected(!attachment.config.isSelected());
    return attachment;
}
```

**Comment by adrian09h** [comment]
@tbarbugli , this pr was fixed as your comments.

File: CHANGELOG.md [hunk 5]
```diff
@@ -27,9 +27,11 @@
 - Catch error body parsing errors
 - Do not show commands list UI when all commands are disabled
 - Renamed `MessageInputClient` to `MessageInputController`
+- Added Large file(20MB) check for uploading file
```
Comment (elevenetc) [inline CHANGELOG.md]: Change log should be updated the latest one.

**Pull request merged**